File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your GPS Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var gps = new Tinkerforge.BrickletGPS(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current coordinates
        gps.getCoordinates(
            function (latitude, ns, longitude, ew, pdop, hdop, vdop, epe) {
                console.log('Latitude: ' + latitude/1000000.0 + ' °');
                console.log('N/S: ' + ns);
                console.log('Longitude: ' + longitude/1000000.0 + ' °');
                console.log('E/W: ' + ew);
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{gps_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your GPS Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let gps = GpsBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current coordinates.
    let get_coordinates_result = gps.get_coordinates().recv()?;

    println!("Latitude: {} °", get_coordinates_result.latitude as f32 / 1000000.0);
    println!("N/S: {}", get_coordinates_result.ns);
    println!("Longitude: {} °", get_coordinates_result.longitude as f32 / 1000000.0);
    println!("E/W: {}", get_coordinates_result.ew);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
